### PR TITLE
Refactor using CommonDelayableHook

### PR DESCRIPTION
**Pull request by ketal178**
Signed-off-by: ketal178 <ketal@qq.com>

# Refactor using CommonDelayableHook  

## 更改内容 / Types of Changes

- [ ] 内核 Core
- [ ] 修复 Bugfix
- [ ] 新功能 New feature
- [x] 增强/优化 Enhancement/optimization
- [ ] 文档 Documentation
- [ ] UI界面 User interface

## 由这个PR解决或关闭的Issues / Issues Fixed or Closed by This PR

-  None

## 检查列表 / Checklist

- [x] 我的代码符合本仓库的代码规范 My code follows the code style of this project
- [ ] 我已经测试了这些更改，它们可以正常工作，且不会破坏任何功能(或我可以解决) I have tested the changes and verified that they work and don't break anything (as well as I can manage).
- [x] 我已合并了对后续工作无意义的commit并确认不会对后续维护造成困扰 I have merged commits that are meaningless for follow-up work and confirmed that they will not cause damage to follow-up maintenance

### CODE REVIEW

File: app/src/main/java/com/rymmmmm/hook/CustomDeviceModel.java [hunk 1]
```diff
@@ -21,35 +21,25 @@
...
-    @Override
-    public int getEffectiveProc() {
-        return SyncUtils.PROC_ANY;
-    }
-
-    @Override
-    public boolean isInited() {
-        return isInit;
+    protected CustomDeviceModel() {
+        super("__NOT_USED__");
```
Comment (cinit) [inline app/src/main/java/com/rymmmmm/hook/CustomDeviceModel.java:38]: `SyncUtils.PROC_ANY`

File: app/src/main/java/com/rymmmmm/hook/CustomMsgTimeFormat.java [hunk 2]
```diff
@@ -27,37 +27,26 @@
...
-    @Override
-    public int getEffectiveProc() {
-        return SyncUtils.PROC_MAIN;
-    }
-
-    @Override
-    public boolean isInited() {
-        return isInit;
+    protected CustomMsgTimeFormat() {
+        super("__NOT_USED__");
```
Comment (cinit) [inline app/src/main/java/com/rymmmmm/hook/CustomMsgTimeFormat.java:45]: `new DexDeobfStep(DexKit.C_TimeFormatterUtils)`

File: app/src/main/java/com/rymmmmm/hook/RemoveMiniProgramAd.java [hunk 3]
```diff
@@ -18,47 +18,36 @@
...
     public static RemoveMiniProgramAd get() {
         return self;
     }
 
+    protected RemoveMiniProgramAd() {
+        super("rq_remove_mini_program_ad");
+    }
+
     @Override
     public int getEffectiveProc() {
```
Comment (cinit) [inline app/src/main/java/com/rymmmmm/hook/RemoveMiniProgramAd.java:44]: 塞super()里

File: app/src/main/java/nil/nadph/qnotified/hook/ChatTailHook.java [hunk 4]
```diff
@@ -285,54 +278,4 @@ public static String getTailRegex() {
...
-
-    @Override
-    public boolean isInited() {
-        return inited;
-    }
-
-    @Override
-    public void setEnabled(boolean enabled) {
-        try {
-            ConfigManager cfg = ExfriendManager.getCurrent().getConfig();
```
Comment (cinit) [inline app/src/main/java/nil/nadph/qnotified/hook/ChatTailHook.java:307]: 看清楚, 这个不一样 `ExfriendManager.getCurrent()`, 所以set/isEnable不要动

File: app/src/main/java/nil/nadph/qnotified/hook/GagInfoDisclosure.java [hunk 5]
```diff
@@ -175,44 +167,4 @@ public int getEffectiveProc() {
         return SyncUtils.PROC_MAIN | SyncUtils.PROC_MSF;
```
Comment (cinit) [inline app/src/main/java/nil/nadph/qnotified/hook/GagInfoDisclosure.java:167]: ` SyncUtils.PROC_MAIN | SyncUtils.PROC_MSF` 塞 super()

File: app/src/main/java/nil/nadph/qnotified/hook/RevokeMsgHook.java [hunk 6]
```diff
@@ -273,45 +266,4 @@ public int getEffectiveProc() {
         //FIXME: is MSF really necessary?
         return SyncUtils.PROC_MAIN | SyncUtils.PROC_MSF;
```
Comment (cinit) [inline app/src/main/java/nil/nadph/qnotified/hook/RevokeMsgHook.java:267]: 塞super()

File: app/src/main/java/nil/nadph/qnotified/hook/SettingEntryHook.java [hunk 7]
```diff
@@ -162,34 +159,13 @@ public void onClick(View v) {
...
-            inited = true;
             return true;
         } catch (Throwable e) {
             log(e);
             return false;
         }
     }
 
-    @Override
-    public boolean checkPreconditions() {
```
Comment (cinit) [inline app/src/main/java/nil/nadph/qnotified/hook/SettingEntryHook.java:174]: 这个 `checkPreconditions() = true` 留着, 不然玄学问题

**Review by cinit — CHANGES_REQUESTED** [review]

**Review by cinit — APPROVED** [review]

**Review by qwq233 — APPROVED** [review]

**Review by NextAlone — APPROVED** [review]

**Pull request merged**